core/services/relay/evm/mercury/transmitter.go

```diff
@@ -192,6 +222,39 @@ func (mt *mercuryTransmitter) HealthReport() map[string]error {
 	return report
 }
 
+func (mt *mercuryTransmitter) runDeleteQueueLoop() {
+	defer mt.wg.Done()
+	runloopCtx, cancel := mt.stopCh.Ctx(context.Background())
+	defer cancel()
+
+	// Exponential backoff for very rarely occurring errors (DB disconnect etc)
+	b := backoff.Backoff{
+		Min:    1 * time.Second,
+		Max:    120 * time.Second,
+		Factor: 2,
+		Jitter: true,
+	}
+
+	for req := range mt.deleteQueue {
+		for {
+			if err := mt.persistenceManager.Delete(runloopCtx, req); err != nil {
```

Comment:
what happens if this queue fills up? what is the impact if some requests are never deleted?

did you consider batching deletes?

Reply:
Yes I considered batching, didn't seem worth the complexity.

If queue fills up, then we don't delete some requests and will try to resend an already sent request on a restart.